go/libraries/doltcore/env/actions/remotes.go

```diff
@@ -136,7 +132,7 @@ func DoPush(ctx context.Context, rsr env.RepoStateReader, rsw env.RepoStateWrite
 		})
 	}
 
-	return nil
+	return err
```

Comment:
This is a little overbroad, I think.

You should check for a specific error type you want to defer before continuing onward. Other error types should be treated as unrecoverable and terminate the method early.